docs/synchronization.md

```diff
@@ -142,13 +144,64 @@ spec:
 
 Examples:
 
-1. [Workflow level semaphore](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-wf-level.yaml)
+1. [Workflow level semaphore](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-wf-level.yamlxb)
 1. [Workflow level mutex](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-mutex-wf-level.yaml)
 1. [Step level semaphore](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-tmpl-level.yaml)
 1. [Step level mutex](https://github.com/argoproj/argo-workflows/blob/main/examples/synchronization-mutex-tmpl-level.yaml)
 
-### Other Parallelism support
+### Multiple synchronization
+
+You can specify multiple mutexes and semaphores to lock in a single workflow or template.
+
+```yaml
+synchronization:
+  mutexes:
+    - name: alpha
+    - name: beta
+  semaphores:
+    - configMapKeyRef:
+        key: foo
+        name: my-config
+    - configMapKeyRef:
+        key: bar
+        name: my-config
+```
+
+The workflow will block until all of these mutexes and semaphores are available.
+
+### Priority
+
+Workflows can have a `priority` set in their specification.
+Workflows with a higher priority value will be queued to take a semaphore or mutex before a lower priority workflow, even if they have been waiting for less time.
+
+!!! Warning
+    A high priority workflow waiting on multiple mutexes or semaphore will make all other workflows which want to acquire those mutexes wait for it to acquire and release all the mutexes or semaphores it is waiting on.
+    This applies even if the lower priority workflows only wish to acquire a subset of those mutexes or semaphores.
```

Comment:
low priority wf won't start running even if the 1 mutex it needs is free? whereas the high priority wf can't even run because it still needs another few mutexes?

Reply:
Correct.

You've made me realise I should document the time ordered queue as well.

Reply:
this seems too restrictive, if the high priority one is unable to run due to multiple locks might as well let the low priority one that just needs 1 lock run

Reply:
i think priority should only apply if all locks are satisfied at the same time, or maybe this could be some kind of new config flag

Reply:
I'm not going to implement that here. It would be a new feature to have a different scheduling system. I'm not even convinced trying to shoehorn complex schedulers into argo-workflows is a good idea. It needs some proper design.

I've updated the documentation to explain more how the queue works.

None of the queuing is new, I've just documented how it worked and how it works in the presence of multiple locks.

Reply:
@tooptoop4 filed a follow-up feature request for this: https://github.com/argoproj/argo-workflows/issues/13630